### PR TITLE
Added full coverage testing support to makefile and updated gitignore…

### PR DESCRIPTION
**Pull request by gabpaw987**
… accordingly again.

Increased unit test coverage of config.cc to 100%.

Please review my changes. Thank you!

### CODE REVIEW

File: config_test.cc [hunk 1]
```diff
@@ -1,40 +1,40 @@
...
 TEST(ServerConfigTest, EmptyConfig) {
   try{
   	ServerConfig config("test_config_4");
-  	EXPECT_EQ(config.GetPortNo(), 80);
   }
   catch (InvalidConfigException &e)
   {
-	std::cout << e.what();
+	EXPECT_EQ(0, strcmp(e.what(), "Port number missing in config file."));
   }
```
Comment (nikhilshridhar) [inline config_test.cc:39]: Are there any test fixtures that we should add? You did mention that we got a 100% test coverage for this class, so I'm assuming that what we have now is good.

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,31 @@
...
+test: CXXFLAGS += -isystem ${GTEST_DIR}/include
+test: SRCFILES += config_parser_test.cc config_test.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a
+test: | clean webserver_test
+
+run:
+	./webserver
+
+run_coverage:
+	./webserver_test
+	gcov -r server.cpp reply.cpp config.cc config_parser.cc
```
Comment (nikhilshridhar) [inline Makefile:20]: Great additions to our Makefile! Now we can our tests and see if it covers everything from the classes and files that we have in this repository.

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,31 @@
...
+SRCFILES = server.cpp reply.cpp config.cc config_parser.cc
+GTEST_DIR = googletest/googletest
+
+all: CXXFLAGS += $(CXXOPTIMIZE)
+all: SRCFILES += main.cpp
+all: webserver
+
+test: CXXFLAGS += -isystem ${GTEST_DIR}/include
+test: SRCFILES += config_parser_test.cc config_test.cc ${GTEST_DIR}/src/gtest_main.cc libgtest.a
+test: | clean webserver_test
```
Comment (nikhilshridhar) [inline Makefile:13]: You should add reply_test.cpp and server_test.cpp as well to include the other unit tests that we have.

File: config_test.cc [hunk 4]
```diff
@@ -1,40 +1,40 @@
...
   try{
   	ServerConfig config("test_config_2");
-  	EXPECT_EQ(config.GetPortNo(), 80);
   }
   catch (PortRangeException &e)
   {
-	std::cout << e.what();
+	EXPECT_EQ(0, strcmp(e.what(), "Port number out of range."));
   }
 }
```
Comment (nikhilshridhar) [inline config_test.cc:21]: Great test cases! We have covered whether the port used for the web server in the config file is out of range or if it is missing. It shows that our web server can handle invalid config files.

File: config_test.cc (hunk 1 above)
Comment (gabpaw987) [reply config_test.cc:39]: I have thought about fixtures but I think it's not really worth the extra code for this small class.

File: Makefile (hunk 3 above)
Comment (gabpaw987) [reply Makefile:13]: Will definitely add that but did not yet know the class names back than. Thank you!

**Comment by gabpaw987** [comment]
Thanks for the review!

**Pull request merged**